arch/x86/boot/compressed/sl_stub.S

```diff
@@ -168,9 +168,27 @@ SYM_FUNC_START(sl_stub)
 	jmp	.Lcpu_setup_done
 
 .Ldo_unknown_cpu:
-	/* AMD is not yet supported */
+	/* Neither Intel nor AMD */
 	ud2
 
+.Ldo_amd:
+	cmpl	$(AMD_CPUID_MFGID_EBX), %ebx
+	jnz	.Ldo_unknown_cpu
+	cmpl	$(AMD_CPUID_MFGID_EDX), %edx
+	jnz	.Ldo_unknown_cpu
+	cmpl	$(AMD_CPUID_MFGID_ECX), %ecx
+	jnz	.Ldo_unknown_cpu
+
+	/* Know it is AMD */
+	movl	$(SL_CPU_AMD), sl_cpu_type(%ebp)
+
+	/*
+	 * Disable maskable interrups in EFLAGS then enable global interrupts
+	 * including SMI and NMI (GIF).
+	 */
+	cli
```

Comment:
Maskable interrupts were already disabled above with the direction flag

Reply:
Fixed